SmartDeviceLink/SDLSecondaryTransportManager.m

```diff
@@ -110,9 +110,15 @@ @interface SDLSecondaryTransportManager ()
 /// The current hmi level of the SDL app.
 @property (strong, nonatomic, nullable) SDLHMILevel currentHMILevel;
 
+/// The current application state of the app on the phone. This information is tracked because TCP transport can only reliably work when the app on the phone is in the foreground. If the phone is locked or the app is put in the background for a long time, the OS can reclaim the socket.
+@property (assign, nonatomic) UIApplicationState currentApplicationState;
```

Comment:
It makes writing tests a lot easier. Any method that `dispatch_sync`s to the main thread can't be called because it freezes the test. This means that we had to swap in a dummy method for each set of tests, which was not very intuitive (you can see how this was done in the deleted code in the the _SDLSecondaryTransportManagerSpec_).